index.js

```diff
@@ -0,0 +1,131 @@
+const {nanoid} = require("nanoid")
+const chalk = require("chalk")
+
+// console.log(chalk.blue("strisng"))
+
+
+const purchases = require("./data/purchases")
+
+function createPurchase (data, name, amount, donation) {
+    data.push({
+        'id' : nanoid(),
+        "name": name || "unidentified item",
+        "amount": Number(amount) || 1, 
+        "donation": Number(donation) || 0
+    })
+    return data
+}
+
+
+function showPurchases(data) {
+    return data.map(x => {
+        return {id : x.id, name: x.name}
```

Comment:
id recommend moving away from 'x' in your callback.  Even el - short for element is better.

perhaps data => purchaseData;  pruchaseData.map(purchase)